sfs/mono/drivingfunction.py

```diff
@@ -667,12 +664,11 @@ def esa_edge_dipole_2d_line(omega, x0, xs, alpha=3/2*np.pi, Nc=None, c=None):
     return -1j*np.pi/alpha * d
 
 
-def _sph_hn2(n, z):
+def _spherical_hn2(n, z):
     """Spherical Hankel function of 2nd kind."""
-    jn, jnp, yn, ynp = sph_jnyn(n, z)
-    return jn - 1j * yn
+    return spherical_jn(n,z) - 1j * spherical_yn(n,z)
 
 
-def _max_order_circular_harmonics(N):
+def _max_order_circular_harmonics(N,Nmax):
     """Compute order of 2D HOA."""
-    return N // 2 if N % 2 == 0 else (N - 1) // 2
+    return Nmax if Nmax else N // 2 if N % 2 == 0 else (N - 1) // 2
```

Comment:
Zeroth order is probably not something many people would want to use, but your logic should still account for it, right?
I think you should check `if Nmax is not None`.